dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/VisitorOpsType.java

```diff
@@ -7,20 +7,24 @@
 import com.dat3m.dartagnan.expression.type.TypeFactory;
 import com.dat3m.dartagnan.parsers.SpirvBaseVisitor;
 import com.dat3m.dartagnan.parsers.SpirvParser;
+import com.dat3m.dartagnan.parsers.program.visitors.spirv.decorations.DecorationType;
+import com.dat3m.dartagnan.parsers.program.visitors.spirv.decorations.Offset;
 import com.dat3m.dartagnan.parsers.program.visitors.spirv.helpers.HelperTags;
 import com.dat3m.dartagnan.parsers.program.visitors.spirv.builders.ProgramBuilder;
 
-import java.util.List;
-import java.util.Set;
+import java.util.*;
+import java.util.stream.IntStream;
 
 public class VisitorOpsType extends SpirvBaseVisitor<Type> {
 
     private static final TypeFactory types = TypeFactory.getInstance();
 
     private final ProgramBuilder builder;
+    private final Offset offset;
 
     public VisitorOpsType(ProgramBuilder builder) {
         this.builder = builder;
+        this.offset = (Offset) builder.getDecorationsBuilder().getDecoration(DecorationType.OFFSET);
```

Comment:
Is this assuming that in SPIRV code the OFFSET decoration is always given? is this true?